packages/mdc-grid-list/foundation.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDCFoundation} from '@material/base';
+import {strings} from './constants';
+
+export default class MDCGridListFoundation extends MDCFoundation {
+  static get strings() {
+    return strings;
+  }
+
+  static get defaultAdapter() {
+    return {
+      getOffsetWidth: () => /* number */ 0,
+      getTileOffsetWidthAtIndex: (/* index: number | null */) => /* number */ 0,
+      setTilesWidth: (/* value: number | null */) => {},
+      registerResizeHandler: (/* handler: EventListener */) => {},
+      deregisterResizeHandler: (/* handler: EventListener */) => {},
+    };
+  }
+  constructor(adapter) {
+    super(Object.assign(MDCGridListFoundation.defaultAdapter, adapter));
+    this.resizeHandler_ = () => this.alignCenter();
```

Comment:
Because this is called every time a window resizes, and that can be very often and rapid if the user is in the process of resizing their window, this has the potential to suffer performance problems. Thus, what you'll want to do is wrap this in a `requestAnimationFrame` and cancel any pending frames if the event handler is called before the previous frame is run.

```js
constructor(adapter) {
  // ...
  this.resizeFrame_ = 0;
}

alignCenter() {
  if (this.resizeFrame_) {
    cancelAnimationFrame(this.resizeFrame_);
  }
  this.resizeFrame_ = this.alignCenter_();
}

alignCenter_() {
  // original alignCenter code
}
```

Doing this ensures that the layout occurring in `alignCenter` always happens at a time when the browser can accommodate it (via `requestAnimationFrame`), and furthermore the `cancelAnimationFrame` ensures that it's never called more times than it needs to be.

Wilson Page's [preventing layout thrashing](http://wilsonpage.co.uk/preventing-layout-thrashing/) article has a great explainer on this ICYI!